core/src/main/scala/kafka/metrics/clientmetrics/ClientMetricsConfig.scala

```diff
@@ -0,0 +1,137 @@
+ * </ul>
+ * For more information please look at kip-714:
+ * https://cwiki.apache.org/confluence/display/KAFKA/KIP-714%3A+Client+metrics+and+observability#KIP714:Clientmetricsandobservability-Clientmetricsconfiguration
+ */
+object ClientMetricsConfig {
+
+  class SubscriptionInfo(subscriptionId: String,
+                         subscribedMetrics: List[String],
+                         pushIntervalMs: Int,
+                         var matchingPatternsList: List[String]) {
+    def getId: String = subscriptionId
+    def getPushIntervalMs: Int = pushIntervalMs
+    def getClientMatchingPatterns: Map[String, String] = ClientMetricsMetadata.parseMatchingPatterns(matchingPatternsList)
+    def getSubscribedMetrics: List[String] = subscribedMetrics
+  }
+
+  object ClientMatchingParams {
+    val CLIENT_ID = "client_id"
+    val CLIENT_INSTANCE_ID = "client_instance_id"
+    val CLIENT_SOFTWARE_NAME = "client_software_name"
+    val CLIENT_SOFTWARE_VERSION = "client_software_version"
+    val CLIENT_SOURCE_ADDRESS = "client_source_address"
+    val CLIENT_SOURCE_PORT = "client_source_port"
+
+    val matchersList = List(CLIENT_ID, CLIENT_INSTANCE_ID, CLIENT_SOFTWARE_NAME,
+                            CLIENT_SOFTWARE_VERSION, CLIENT_SOURCE_ADDRESS, CLIENT_SOURCE_PORT)
+
+    def isValidParam(param: String) = matchersList.contains(param)
+  }
+
+  object ClientMetrics {
+    // Properties that are used to create the subscription for client_metrics.
+    val SubscriptionMetrics = "metrics"
+    val PushIntervalMs = "interval.ms"
+    val ClientMatchPattern = "match"
+
+    val DEFAULT_PUSH_INTERVAL = 30 * 1000 // 5 minutes
+
+    // Definitions of accepted values
+    val configDef = new ConfigDef()
+      .define(SubscriptionMetrics, LIST, "", MEDIUM, "List of the subscribed metrics")
+      .define(ClientMatchPattern, LIST, "", MEDIUM, "Pattern used to find the matching clients")
+      .define(PushIntervalMs, INT, DEFAULT_PUSH_INTERVAL, MEDIUM, "Interval that a client can push the metrics")
+
+    def names = configDef.names
+
+    def validate(subscriptionId :String, properties :Properties): Unit = {
+      if (subscriptionId.isEmpty) {
+        throw new InvalidRequestException("subscriptionId can't be empty")
+      }
+      validateProperties(properties)
+    }
+
+    def validateProperties(properties: Properties) = {
+      // Make sure that all the properties are valid.
+      properties.keySet().forEach(x =>
+        if (!names.contains(x))
+          throw new InvalidRequestException(s"Unknown client metric configuration: $x")
+      )
+
+      // Make sure that push interval is between 100ms and 1 hour.
+      if (properties.containsKey(PushIntervalMs)) {
+        val pushIntervalMs = Integer.parseInt(properties.getProperty(PushIntervalMs))
+        if (pushIntervalMs < 100 || pushIntervalMs > 3600000)
+          throw new InvalidRequestException(s"Invalid parameter ${PushIntervalMs}")
```

Comment:
This exception would not mention which parameter was invalid. It would say something like `Invalid parameter 100`. Probably ought to improve it.

Reply:
Changed the message to capture values, please check.